ceph/rados/pool_workflows.py

```diff
@@ -134,12 +150,14 @@ def fill_omap_entries(self, pool_name, **kwargs):
             deep_scrub_status = self.scrub_obj.verify_scrub_deepscrub(
                 before_deep_scrub_log, after_deep_scrub_log, "deepscrub"
             )
+
             if deep_scrub_status:
-                log.debug(
-                    "deep scrubbing has not been completed on the pool"
-                    "Sleeping for 10 seconds and checking again"
-                )
-                time.sleep(10)
+                log.info("deep scrubbing has not been completed on the pool")
+                if check_omap_entires():
```

Comment:
If omap entries are found here, we are breaking out of the loop. Then again , line : 174 check seems redundant. We are rebooting the primary again post confirmation. can be removed.

Reply:
The check at line 174 is not entirely redundant as it takes care of this edge case:
Consider that deep-scrub got completed, and while it was in progress, we restarted the primary OSD multiple times but OMAP never showed up, so now we will be out of the while loop which checks for completion of deep-scrub and also OMAP entries were never found, so this check just covers this worst possible edge case

In a happy path workflow, yes this is redundant

So I would request we let it be there, it won't do any harm as the code has been modified to ensure OSD is restarted only if OMAPs are not available

Reply:
Sure. Just add a debug statement somewhere in next PR, so that we can check, in how many cases, was this actually required.